test/EFCore.PG.FunctionalTests/Query/SimpleQueryNpgsqlTest.Where.cs

```diff
@@ -8,93 +8,9 @@ namespace Npgsql.EntityFrameworkCore.PostgreSQL.Query
-            AssertContainsSqlFragment("WHERE DATE_TRUNC('day', NOW()) ");
-        }
-
-        public override async Task Where_datetime_date_component(bool isAsync)
-        {
-            await base.Where_datetime_date_component(isAsync);
-            AssertContainsSqlFragment("WHERE DATE_TRUNC('day', o.\"OrderDate\")");
-        }
-
-        public override async Task Where_datetime_year_component(bool isAsync)
-        {
-            await base.Where_datetime_year_component(isAsync);
-            AssertContainsSqlFragment("DATE_PART('year', o.\"OrderDate\")");
-        }
-
-        public override async Task Where_datetime_month_component(bool isAsync)
-        {
-            await base.Where_datetime_month_component(isAsync);
-            AssertContainsSqlFragment("DATE_PART('month', o.\"OrderDate\")");
-        }
-
-        public override async Task Where_datetime_dayOfYear_component(bool isAsync)
-        {
-            await base.Where_datetime_dayOfYear_component(isAsync);
-            AssertContainsSqlFragment("DATE_PART('doy', o.\"OrderDate\")");
-        }
-
-        public override async Task Where_datetime_day_component(bool isAsync)
-        {
-            await base.Where_datetime_day_component(isAsync);
-            AssertContainsSqlFragment("DATE_PART('day', o.\"OrderDate\")");
-        }
-
-        public override async Task Where_datetime_hour_component(bool isAsync)
-        {
-            await base.Where_datetime_hour_component(isAsync);
-            AssertContainsSqlFragment("DATE_PART('hour', o.\"OrderDate\")");
-        }
-
-        public override async Task Where_datetime_minute_component(bool isAsync)
-        {
-            await base.Where_datetime_minute_component(isAsync);
-            AssertContainsSqlFragment("DATE_PART('minute', o.\"OrderDate\")");
-        }
-
-        public override async Task Where_datetime_second_component(bool isAsync)
-        {
-            await base.Where_datetime_second_component(isAsync);
-            AssertContainsSqlFragment("DATE_PART('second', o.\"OrderDate\")");
-        }
-
         [Theory(Skip = "SQL translation not implemented, too annoying")]
         public override Task Where_datetime_millisecond_component(bool isAsync) => null;
 
-        [Theory]
-        [MemberData(nameof(IsAsyncData))]
-        public async Task Where_datetime_dayOfWeek_component(bool isAsync)
-        {
-            await AssertQuery<Order>(
-                isAsync,
-                oc => oc.Where(o =>
-                    o.OrderDate.Value.DayOfWeek == DayOfWeek.Tuesday),
-                entryCount: 168);
-            AssertContainsSqlFragment("WHERE CAST(FLOOR(DATE_PART('dow', o.\"OrderDate\")) AS int)");
-        }
-
-        public override async Task Where_string_indexof(bool isAsync)
-        {
-            await base.Where_string_indexof(isAsync);
-            AssertContainsSqlFragment("WHERE ((STRPOS(c.\"City\", 'Sea') - 1) <> -1)");
-        }
-
```

Comment:
With so much of this class removed, any reason not to move the leftovers over to `SimpleQueryNpgsqlTest`? If nothing else, `Convert_ToString` probably shouldn't be here..

Reply:
The main reason is the fact that 3.0.0 no longer does client evaluation.

2.2 did client evaluation, so it was important for us to have SQL assertions to make sure translations were actually working - that's no longer the case now(*). At the same time, SQL assertions create a lot of work, since minor changes in SQL generation upstream cause our tests to break (e.g. parentheses are added or similar stuff). As a result, I've shifted the approach to only do SQL assertions in specific scenarios, i.e. where Npgsql generates different SQL from what is generated by default upstream, or where we did some special translation that is worth "documenting" via a test. That's why a lot of test overrides have disappeared.

Let me know if you think that's wrong in any way, or if you think there are specific tests where we should have SQL assertions, etc.

(*) 3.0 does do client evaluation for top-level projections (i.e. if the Select is the last thing). So tests which exercise the translation in the last Select are problematic, and should either do SQL assertions, or be rewritten to do the translation in a Where clause instead.

Reply:
Ahh. I think I was asking a different question, but I much prefer this answer. I misunderstood  why these tests were changing, but your explanation makes sense.

Not having SQL asserts everywhere has me a bit nervous that we could (theoretically?) be missing some syntax nuance, but I appreciate the allure of not having to fix broken tests every time there's upstream changes.

You've been living and breathing this port for the last several days, so if you're comfortable with this approach, it sounds good to me.

Reply:
Well, the base class already asserts that the results coming out of the database match the expected ones - it runs the same LINQ in memory and compares the results. So we know the provider is returning the correct results.

The added value beyond that of asserting on SQL seems quite limited, given that client evaluation has gone away. FYI the EF team themselves don't do SQL asserts in their Sqlite provider. Some cases do warrant it, but I don't think we should just do it everywhere - the number of times I've had to just fix SQL because of some minor change makes me want to avoid that...